### DISCUSSION TITLE
Generate FIPs/README.md based on FIPs and FRCs

### DISCUSSION THREAD

**Discussion by Kubuxu** [filecoin-project/FIPs#698] [Enhancements - Organizational]
Currently, the list of FIPs in readme is manually maintained. This leads to multiple FIP documents having an invalid YAML header (see https://github.com/filecoin-project/FIPs/pull/696) or the header not reflecting the current state of the FIP (see https://github.com/filecoin-project/FIPs/pull/697).

It would be best if the README was generated from FIPs and updated automagically. This way, there is only one source of truth regarding what state the FIP is in, and the FIP would need a valid YAML header to be included in the list.

**Comment by bmann** [comment]
I wrote this code previously. You could make a github action that builds the site and then overwrites the README with an auto generated one, as part of the build process for Jekyll.

The other thing to do is to not actually list it in the README at all, and just link to the auto-generated table from the HTML output of building the Jekyll site.

**Reply by anorth** [reply]
I think this should be independent of any web publishing, so the raw README in the repo is always up to date. A Makefile rule plus a CI action that checks that process has actually been run (i.e. no diff when run again) could suffice.
